docs/production/runtime_variables.md

```diff
@@ -0,0 +1,163 @@
+# Runtime Variables
+Runtime Variables are a set of global variables that can be used by every block. These are useful for storing constants shared by multiple blocks or constants whose value is determined at pipeline runtime (hence _runtime_ variables).
+
+## Using Runtime Variables
+
+Runtime Variables can be accessed via the `**kwargs` parameter in your block function.
+
+```python
+from pandas import DataFrame
+from os import path
+
+if 'data_loader' not in globals():
+    from mage_ai.data_preparation.decorators import data_loader
+
+
+@data_loader
+def loader_data(**kwargs) -> DataFrame:
+    filepath = kwargs.get('filepath')
+    row_limit = kwargs.get('row_limit')
+    return pd.read_csv(filepath, nrows=row_limit)
+```
+
+Currently, runtime variables must be of primitive Python types or some basic containers:
+- integer
+- string
+- float
+- boolean
+- list
+- dictionary
+- set
+
+Runtime variable names must be a valid Python identifier (i.e., no spaces in name, can't start with a number).
+
+## Creating Runtime Variables In-Editor
+
+**_WIP_**
+
+## Running Pipeline with Runtime Variables
+
+If your pipeline is configured to use runtime variables, you can still execute your pipeline outside the code editor. Provide the runtime variables as keyword arguments to `mage_ai.run()`:
+
+```python
+mage_ai.run('sample_pipeline', 'repos/default_repo', filepath = 'path/to/my/file.csv', row_limit=1000)
+```
+
+## Example - Aggregating Daily Logs
+A common ETL task is processing and analyzing daily events, or in the case of this example, logs from a web application.
+
+| log_date            | type    | source    |
+| ------------------- | ------- | --------- |
+| 2022-07-27T20:19:20 | INFO    | react-1   |
+| 2022-07-27T19:18:45 | WARNING | express-2 |
+| 2022-07-27T07:20:26 | ERROR   | express-1 |
+| 2022-07-27T00:42:37 | ERROR   | react-1   |
+| 2022-07-26T16:35:28 | DEBUG   | react-1   |
+| 2022-07-26T10:19:32 | INFO    | express-1 |
+
+Suppose we want to know the distribution of log types at the end of every day. Using Mage's runtime variables this is made a very simple task:
+1. Provide the current date to partition the log file using **runtime variables**
+2. Calculate the distribution of log types for the current date
+
+We'll define the transformer below to perform this statistics calculation:
```

Comment:
We usually only load events of the given date partition into memory instead of loading all the events into memory and filtering them.

Reply:
I see, I'm assuming this is when loading from a database or warehouse rather than a log file, correct?

Reply:
This could be either from data warehouse, or log files. For log files, we usually save hourly logs or daily logs in a file.

Reply:
I'll take a look at some of the log files and rebuild this example.

Reply:
Updated example to now load all log files modified on the given date, and use that to aggregate statistics.